### PR TITLE
Update Flutter Gallery to use new asset api for package assets

### LINKED ISSUE #12199: flutter gallery should use new package option when pulling assets
flutter gallery uses assets from a package. It would be good to update it to use https://flutter.io/assets-and-images/#from-packages

### PR DESCRIPTION
**Pull request by szakarias**
Fixes #12199.

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/pesto_demo.dart [hunk 1]
```diff
@@ -521,7 +535,9 @@ class Recipe {
   final String author;
   final String description;
   final String imagePath;
+  final String imagePackage;
   final String ingredientsImagePath;
+  final String ingredientsImagePackage;
```
Comment (mravn-google) [inline examples/flutter_gallery/lib/demo/pesto_demo.dart]: [nit] I guess we could use the same package for both `image` and `ingredientsImage`.
Your call.

**Review by mravn-google — APPROVED** [review]
LGTM

We now specify a single logical asset using two items, the package name and the asset name. I wonder if a simple `PackageAsset` class with these two fields and a `const` constructor would make the code more readable.

**Comment by mit-mit** [comment]
lgtm

File: examples/flutter_gallery/lib/demo/pesto_demo.dart (hunk 1 above)
Comment (szakarias) [reply examples/flutter_gallery/lib/demo/pesto_demo.dart]: I'll leave it as in theory the packages could be different.

**Review by szakarias — COMMENTED** [review]
I see your point about the two items for one logical asset. However I also think it sends a mixed signal introducing a utility class here when we decided not to have one in the API . As discussed, I'll leave it like this.

**Pull request merged**